modules/access-management/main.tf

```diff
@@ -0,0 +1,147 @@
+      "tags"              = var.privileged_compute_ag_tags
+      "ibm_ids"           = var.privileged_compute_ag_ibm_ids
+      "service_ids"       = var.privileged_compute_ag_service_ids
+      }, {
+      # observer observability
+      "access_group_name" = var.observer_observability_ag_name
+      "policies"          = var.observer_observability_ag_policies
+      "dynamic_rules"     = var.observer_observability_ag_dynamic_rules
+      "add_members"       = var.observer_observability_ag_add_members
+      "description"       = var.observer_observability_ag_description
+      "tags"              = var.observer_observability_ag_tags
+      "ibm_ids"           = var.observer_observability_ag_ibm_ids
+      "service_ids"       = var.observer_observability_ag_service_ids
+      }, {
+      # observer security
+      "access_group_name" = var.observer_security_ag_name
+      "policies"          = var.observer_security_ag_policies
+      "dynamic_rules"     = var.observer_security_ag_dynamic_rules
+      "add_members"       = var.observer_security_ag_add_members
+      "description"       = var.observer_security_ag_description
+      "tags"              = var.observer_security_ag_tags
+      "ibm_ids"           = var.observer_security_ag_ibm_ids
+      "service_ids"       = var.observer_security_ag_service_ids
+      }, {
+      # observer network
+      "access_group_name" = var.observer_network_ag_name
+      "policies"          = var.observer_network_ag_policies
+      "dynamic_rules"     = var.observer_network_ag_dynamic_rules
+      "add_members"       = var.observer_network_ag_add_members
+      "description"       = var.observer_network_ag_description
+      "tags"              = var.observer_network_ag_tags
+      "ibm_ids"           = var.observer_network_ag_ibm_ids
+      "service_ids"       = var.observer_network_ag_service_ids
+      }, {
+      # observer compute
+      "access_group_name" = var.observer_compute_ag_name
+      "policies"          = var.observer_compute_ag_policies
+      "dynamic_rules"     = var.observer_compute_ag_dynamic_rules
+      "add_members"       = var.observer_compute_ag_add_members
+      "description"       = var.observer_compute_ag_description
+      "tags"              = var.observer_compute_ag_tags
+      "ibm_ids"           = var.observer_compute_ag_ibm_ids
+      "service_ids"       = var.observer_compute_ag_service_ids
+    }
+  ]
+}
+
+module "access_groups" {
+  for_each          = { for obj in local.ag_list : obj.access_group_name => obj }
+  source            = "../.."
+  access_group_name = each.value.access_group_name
+  policies          = each.value.policies
+  dynamic_rules     = each.value.dynamic_rules
+  add_members       = each.value.add_members
+  description       = each.value.description
+  tags              = each.value.tags
+  ibm_ids           = each.value.ibm_ids
+  service_ids       = each.value.service_ids
+}
+
+resource "ibm_iam_custom_role" "custom_iam_roles" {
```

Comment:
I think I raised this before, but what if one of these custom roles need to be included in one of the access groups?

Reply:
as per discussions we can remove custom role support for first release of submodule

Reply:
Removed custom roles from the submodule